### ISSUE TITLE
Seeking guidance on reproducing paper results with provided config

### ISSUE THREAD

**Issue by MagicTZ** [3DAgentWorld/OpenGS-SLAM#6]
Hi, thanks for your excellent work!

I’m trying to reproduce the results from the paper using the default base_config.yaml provided in the repository, but my results **(test in 100613.yaml)** are noticeably worse.

![Image](https://github.com/user-attachments/assets/3eedb707-7913-448d-8555-4945be1ded5d)

![Image](https://github.com/user-attachments/assets/aab06766-75f8-4626-884d-85601e111d5d)

```
{
    "rmse": 0.4715244245282105,
    "mean": 0.43195006852728046,
    "median": 0.370154700846674,
    "std": 0.18908839527040722,
    "min": 0.08403950664170633,
    "max": 0.9655146783936671,
    "sse": 7.114729053653122
}

{
    "mean_psnr": 22.75845586603338,
    "mean_ssim": 0.7847526105967435,
    "mean_lpips": 0.41799326412605514
}
```

Could you kindly:

1. _Confirm whether the provided config.yaml is the exact configuration used in the paper?_
2. _Share any additional experimental details (e.g., hyperparameters, data preprocessing, random seeds) that might affect reproducibility?_

For reference, here’s the base_config.yaml I used:
```yaml
Results:
  save_results: True
  save_dir: "results"
  save_trj: True
  save_trj_kf_intv: 10
  use_gui: False
  eval_rendering: True
  use_wandb: False
  color_refinement: True

Dataset:
  type: 'waymo'
  sensor_type: 'monocular'
  pcd_downsample: 64
  pcd_downsample_init: 32
  adaptive_pointsize: True
  point_size: 0.01

Training:
  init_itr_num: 1050
  init_gaussian_update: 100
  init_gaussian_reset: 500
  init_gaussian_th: 0.005
  init_gaussian_extent: 30
  tracking_itr_num: 100
  mapping_itr_num: 150
  gaussian_update_every: 150
  gaussian_update_offset: 50
  gaussian_th: 0.7
  gaussian_extent: 1.0
  gaussian_reset: 2001
  size_threshold: 20
  kf_interval: 5
  window_size: 8
  pose_window: 3
  edge_threshold: 1.1
  rgb_boundary_threshold: 0.01

  kf_translation: 0.08
  kf_min_translation: 0.05
  kf_overlap: 0.9
  kf_cutoff: 0.3
  prune_mode: 'slam'
  single_thread: False
  spherical_harmonics: False

  lr:
    cam_rot_delta: 0.003
    cam_trans_delta: 0.001

opt_params:
  iterations: 30000
  position_lr_init: 0.0016
  position_lr_final: 0.0000016
  position_lr_delay_mult: 0.01
  position_lr_max_steps: 30000
  feature_lr: 0.0025
  opacity_lr: 0.05
  scaling_lr: 0.001
  rotation_lr: 0.001
  percent_dense: 0.01
  lambda_dssim: 0.2
  densification_interval: 100
  opacity_reset_interval: 3000
  densify_from_iter: 500
  densify_until_iter: 15000
  densify_grad_threshold: 0.0002

model_params:
  sh_degree: 0
  source_path: ""
  model_path: ""
  resolution: -1
  white_background: False
  data_device: "cuda"

pipeline_params:
  convert_SHs_python: False
  compute_cov3D_python: False

```

**Comment by Ysc-shark** [comment]
Thanks for your interest in our work.

I test it on my RTX 3090 server using default setting, and it shows a similar preformance to the paper:
![Image](https://github.com/user-attachments/assets/785ad568-aa67-4051-969c-82aa705997de)
<img width="680" alt="Image" src="https://github.com/user-attachments/assets/20392cd8-92e8-4752-906c-1e932b097c20" />

During our working process, we do find outcomes fluctuating due to some random factors such as downsampling of pointmaps. Unfortunately, we didn't  record the random seed. 
However, this degree of decline on rendering quality really range out of our expectations.
We also find that, if your device is crowded, the result may be affected seriously.

The following advice may lead you to reproduce the results in paper:

1. Make sure your device is free (especially gpu), and try it again.
2. Change this line from 3 to 1 to improve stability. https://github.com/3DAgentWorld/OpenGS-SLAM/blob/a6822c9513042963295c30882be7e50388f85fd1/utils/slam_backend.py#L263

This is the result I run after conduct this change:
![Image](https://github.com/user-attachments/assets/50d2e4a0-04a9-4bdf-b82b-b8813e61c89b)

**Comment by MagicTZ** [comment]
Thank you for your reply. However, regarding the tuning of the parameter `prune_coviz`, I found that the value from 3 to 1 rather made the results worse. My understanding is that this parameter relaxes the criteria for pruning, which retains more point clouds but may result in more noise appearing.
I'm not sure if I'm understanding this correctly. Another thing is about the downsampling parameter of pointmap, are you using 64 and 32 now as well?

**Comment by Ysc-shark** [comment]
Firstly, your understanding is correct, I give this advice due to relaxing the criteria for pruning lead to more stable tracking during crush turn, which I learned from the experiment of our new work. However, based on your response, it seems to be unhelpful for 100613.

Secondly, I have checked your config, we are using the same hyperparameters including downsampling rate. Besides this, you may try to re-activate the following code, and see whether it could help:
https://github.com/3DAgentWorld/OpenGS-SLAM/blob/a6822c9513042963295c30882be7e50388f85fd1/utils/slam_backend.py#L316-L321

If it still perform badly, the effect of machine may be considered as a key factor. It really confuse me because we can reproduce similar results on both our 3090 and A6000 server, and this degree of decline of rendering quality is beyond our expectation on machine gap. 

Could you tell me your device? Whether there is resource conflict with other programs?

If anyone else meet this problem, it would be meaningful to share your result values and device.

**Comment by MagicTZ** [comment]
Thanks for your quick reply. My experiments are conducted on RTX4090 and Ubuntu20.04.

**Comment by mohhef** [comment]
would be interested to know as well, getting ATE of 2.~ 3090 GPU, ubuntu 20.04, default everything

<img width="626" height="478" alt="Image" src="https://github.com/user-attachments/assets/5729be8a-64db-48f3-b3fb-0ebe6cdbbb09" />

```
OpenGS-SLAM: Evaluating ATE at frame:  116
Eval: RMSE ATE [m] 0.2689277190435625
OpenGS-SLAM: new keyframe:  122
OpenGS-SLAM: new keyframe:  129
OpenGS-SLAM: Resetting the opacity of non-visible Gaussians
OpenGS-SLAM: new keyframe:  136
OpenGS-SLAM: new keyframe:  143
OpenGS-SLAM: new keyframe:  150
OpenGS-SLAM: new keyframe:  156
OpenGS-SLAM: new keyframe:  162
OpenGS-SLAM: new keyframe:  168
OpenGS-SLAM: new keyframe:  176
OpenGS-SLAM: new keyframe:  192
OpenGS-SLAM: Evaluating ATE at frame:  193
Eval: RMSE ATE [m] 1.007317994100342
Eval: RMSE ATE [m] 1.0085276128168947
Eval: Total time 606.1309375
Eval: Total FPS 0.3266620918850558
Eval: RMSE ATE [m] 1.0085276128168947
average render time is: 0.7141208934212873
```

It increases by a lot after the evaluation at 129
